geowebcache/core/src/main/java/org/geowebcache/config/XMLConfiguration.java

```diff
@@ -266,13 +268,13 @@ public void setDefaultValues(TileLayer layer) {
             URL proxyUrl = null;
             try {
                 if (getGwcConfig().getProxyUrl() != null) {
-                    proxyUrl = new URL(getGwcConfig().getProxyUrl());
+                    proxyUrl = new URI(getGwcConfig().getProxyUrl()).toURL();
                     log.fine("Using proxy " + proxyUrl.getHost() + ":" + proxyUrl.getPort());
```

Comment:
If I guess correctly, this change is related to the deprecation or "new URL(string)" [in Java 21](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/net/URL.html#%3Cinit%3E(java.lang.String)).

Rather than repeating this pattern over and over, along with the exception rethrowing, could we have a dedicated support method, similar (but not same) to what we have URLs.fileToURL() in GeoTools?

Reply:
You got it.